scipy/io/mmio.py

```diff
@@ -248,16 +248,16 @@ def info(self, source):
             while not line.strip():
                 line = stream.readline()
 
-            line = line.split()
+            split_line = line.split()
             if format == self.FORMAT_ARRAY:
-                if not len(line) == 2:
+                if not len(split_line) == 2:
                     raise ValueError("Header line not of length 2: " + line)
```

Comment:
Before this change ValueError is reported to raise `TypeError: cannot concatenate 'str' and 'list' objects`

I didn't validate this, but it's seems clear to me that it would (line would be an list from calling split)